crypto/evp/pkey_mac.c

```diff
@@ -81,7 +82,8 @@ static int pkey_mac_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
     sctx = EVP_PKEY_CTX_get_data(src);
     dctx = EVP_PKEY_CTX_get_data(dst);
 
-    if (!EVP_MAC_CTX_copy(dctx->ctx, sctx->ctx))
+    if (sctx->ctx->data != NULL
+        && !dctx->ctx->meth->dup(dctx->ctx->data, sctx->ctx->data))
```

Comment:
Shouldn't this be this?

``` C
    EVP_MAC_CTX_free(dctx->ctx);
    if ((dctx->ctx = EVP_MAC_CTX_dup(sctx->ctx)) == NULL) 
```

Reply:
That dctx->ctx->meth->dup() function is really a copy function that I renamed by mistake. But that would work too.

Reply:
The pkey_mac_init() above just allocated it, directly calling free on it doesn't make much sense.